.github/workflows/docs.yml

```diff
@@ -117,14 +117,30 @@ jobs:
           echo "Benchmarks site DONE."
 
           # Stage and commit benchmark results
+          # 1. Create a new branch and commit benchmark results
           echo "Staging and committing benchmark results..."
+          git checkout -b "benchmark-${TAG}"
           git add .asv/results/
           git add docs/benchmarks/
           git commit -m "Add benchmark results for ${TAG}"
           echo "Benchmark results committed."
-          git push origin master
+
+          # 2. Push to the new branch
+          echo "Pushing benchmark results to new branch..."
+          git push -u origin "benchmark-${TAG}"
+
+          # 3. Create a PR
+          PR_URL="$(gh pr create --title "chore: Add benchmark results for ${TAG}" --body "Add benchmark results for ${TAG}" --base master)"
+          echo "PR URL: ${PR_URL}"
+
+          # 4. Merge the PR
+          gh pr merge "${PR_URL}" --admin --squash --delete-branch
```

Comment:
Using `gh pr create` and `gh pr merge` to push the changes to `master`, as seen [here](https://github.com/krlmlr/pr-bots/blob/main/.github/workflows/write-file-branch-merge.yaml)